src/app/data-model-provider/Provider.h

```diff
@@ -79,6 +79,15 @@ class Provider : public ProviderMetadataTree
     ///    - validation of ACL/timed interaction flags/writability, if those checks are desired.
     virtual ActionReturnStatus WriteAttribute(const WriteAttributeRequest & request, AttributeValueDecoder & decoder) = 0;
 
+    ///   Indicates the start/end of a series of list operations. This function will be called either before the first
+    ///   Write operation or after the last one of a series of consecutive attribute data of the same attribute.
+    ///
+    ///   1) This function will be called if the client tries to set a nullable list attribute to null.
+    ///   2) This function will only be called for a series of consecutive attribute data (regardless the kind of list
+    ///   operation) of the same attribute at the beggining or end of the Write operations.
+    ///   3) The opType argument indicates the type of notification (Start, Failure, Success).
+    virtual void ListAttributeWriteNotification(const ConcreteAttributePath & aPath, BitFlags<ListWriteOperation> opType) = 0;
```

Comment:
Why is the second arg a BitFlags?  Only one of the value is allowed a time, and worse yet the set of values 0, 1, 2 is not a valid set of values for an enum that's used in BitFlags.

This should just be of type `ListWriteOperation`, no?

Reply:
Thanks for noticing, I think that this was a "vestige" of a previous attempt that I was working in and got carried away before the PR, I feel a little silly right now. Yes, this should be a plain ListWriteOperation enum since there is no need for BitFlags right now.